spec/lib/service/steps_inspector_spec.rb

```diff
@@ -325,8 +309,32 @@ def might_raise
       end
 
       it "returns an error related to the exception" do
-        expect(error).to match(/RuntimeError: BOOM/)
+        expect(error).to match(/BOOM \([^(]*RuntimeError[^)]*\)/)
       end
     end
   end
+
+  describe "#inspect" do
+    let(:parameter) { nil }
+
+    it "outputs the service class name, the steps results and the specific error" do
+      expect(inspector.inspect.gsub(%r{ \(\d+\.\d+ ms\)}, "")).to eq(<<~OUTPUT)
+        Inspecting DummyService result object:
+
+        [ 1/10] [options] default ✅
+        [ 2/10] [model] model ✅
+        [ 3/10] [policy] policy ✅
+        [ 4/10] [params] default ❌
+
+        (6 more steps not shown as the execution flow was stopped before reaching them)
+
+
```

Comment:
super detail here, but why not just one linebreak?

Reply:
I thought it would provide a better separation between the steps inspection part and the explanation part. But we could just use one blank line instead, I don’t have a strong opinion on this.